server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -327,6 +331,8 @@ public Query getFieldQuery(String field, String queryText, boolean quoted) throw
             return getFieldQuery(field, queryText, getPhraseSlop());
         }
 
+        onDiscoveredField(field);
```

Comment:
Probably too early, should be added after `if (field != null) {` below

Reply:
done